### PR TITLE
[DataGrid] Fix server-side pagination

### LINKED ISSUE #568: [DataGrid] server side pagination not working
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

Second page not loading

## Expected Behavior 🤔

Pagination should work correct

## Steps to Reproduce 🕹

https://material-ui.com/components/data-grid/pagination/#server-side-pagination

Steps:

1. Click next page
2.
3.
4.

### PR DESCRIPTION
**Pull request by dtassone**
Fix #568

### CODE REVIEW

**Review by oliviertassinari — CHANGES_REQUESTED** [review]
> fix #568 and other small issues

Please one pull request at the time, it doesn't work https://deploy-preview-639--material-ui-x.netlify.app/components/data-grid/pagination/#server-side-pagination

**Review by oliviertassinari — CHANGES_REQUESTED** [review]
Pagination is not working: https://deploy-preview-639--material-ui-x.netlify.app/components/data-grid/pagination/#server-side-pagination

**Comment by dtassone** [comment]
> Pagination is not working: https://deploy-preview-639--material-ui-x.netlify.app/components/data-grid/pagination/#server-side-pagination. The CI should be red, a test case should be falling in the current state.

I tried the docs locally, and it works. I also tried with the filtering, columns, and all the new options disabled, and it works.
The error on the server is 
image

Any idea? 
🤔

**Comment by oliviertassinari** [comment]
> I tried the docs locally, and it works. I also tried with the filtering, columns, and all the new options disabled, and it works.

Is it feature flag related? 

> The error on the server is

This is a client side error, and unrelated.

**Comment by dtassone** [comment]
> > I tried the docs locally, and it works. I also tried with the filtering, columns, and all the new options disabled, and it works.
> 
> Is it feature flag related?
> 
> > The error on the server is
Yeah I'm wondering... Let me push an update to test that
> 
> This is a client side error, and unrelated.

I mean, it's the error on the server, I didn't mean it occured on the backend

**Comment by oliviertassinari** [comment]
> I mean, it's the error on the server, I didn't mean it occured on the backend

I don't follow, we probably have a different vocabulary, what's the backend, what's the server? I was assuming this terminology: server = node.js, backend = we don't have one.

**Comment by dtassone** [comment]
> > I mean, it's the error on the server, I didn't mean it occured on the backend
> 
> I don't follow, we probably have a different vocabulary, what's the backend, what's the server? I was assuming this terminology: server = node.js, backend = we don't have one.

For me, server can be the CI server, vs local where it's my own machine.
Backend as opposed to frontend, the code that runs on the server machine, so nodejs...

**Comment by dtassone** [comment]
I managed to reproduce the error locally. It's due to the production code, so probably an issue with the minifier... 🤔

File: packages/grid/_modules_/grid/models/gridOptions.tsx [hunk 1]
```diff
@@ -312,10 +312,10 @@ export const DEFAULT_GRID_OPTIONS: GridOptions = {
...
   sortingOrder: ['asc', 'desc', null],
   columnTypes: DEFAULT_COLUMN_TYPES,
-  disableColumnMenu: process.env.NODE_ENV === 'production',
-  disableColumnFilter: process.env.NODE_ENV === 'production',
-  disableColumnSelector: process.env.NODE_ENV === 'production',
-  hideToolbar: process.env.NODE_ENV === 'production',
+  // disableColumnMenu: process.env.NODE_ENV === 'production',
+  // disableColumnFilter: process.env.NODE_ENV === 'production',
+  // disableColumnSelector: process.env.NODE_ENV === 'production',
+  // hideToolbar: process.env.NODE_ENV === 'production',
```
Comment (DanailH) [inline packages/grid/_modules_/grid/models/gridOptions.tsx:318]: I guess this is only for testing and before merging these will be uncommented.

**Comment by DanailH** [comment]
Just a general question, a lot of the changes from this PR are also in this one [#642](https://github.com/mui-org/material-ui-x/pull/642). I guess this PR should be the first one merged right?

File: packages/grid/_modules_/grid/models/gridOptions.tsx (hunk 1 above)
Comment (dtassone) [reply packages/grid/_modules_/grid/models/gridOptions.tsx:318]: Yes there is an issue with the CI and the docs so I was checking if that caused it. I will rollback and apply the experimental flag from @oliviertassinari

**Comment by dtassone** [comment]
> Just a general question, a lot of the changes from this PR are also in this one [#642](https://github.com/mui-org/material-ui-x/pull/642). I guess this PR should be the first one merged right?

Yes

**Review by oliviertassinari — APPROVED** [review]

**Comment by oliviertassinari** [comment]
The documentation page are broken after the changes, however, it doesn't seem directly related, so happy to move forward.

**Pull request merged**